letsencrypt/scripts/main.py

```diff
@@ -9,61 +9,100 @@
-                             "changes.")
-    parser.add_argument("-r", "--redirect", dest="redirect",
-                        action="store_const", const=True,
-                        help="Automatically redirect all HTTP traffic to HTTPS "
-                             "for the newly authenticated vhost.")
-    parser.add_argument("-n", "--no-redirect", dest="redirect",
-                        action="store_const", const=False,
-                        help="Skip the HTTPS redirect question, allowing both "
-                             "HTTP and HTTPS.")
-    parser.add_argument("-e", "--agree-tos", dest="eula", action="store_true",
-                        help="Skip the end user license agreement screen.")
-    parser.add_argument("-t", "--text", dest="use_curses", action="store_false",
-                        help="Use the text output instead of the curses UI.")
-    parser.add_argument("--test", dest="test", action="store_true",
-                        help="Run in test mode.")
+    add = parser.add_argument
+
+    add("-d", "--domains", metavar="DOMAIN", nargs="+")
+    add("-s", "--acme-server", "--server", default="letsencrypt-demo.org:443",
+        help="CA hostname (and optionally :port). The server certificate must "
+             "be trusted in order to avoid further modifications to the "
+             "client.")
+
+    add("-p", "--privkey", type=read_file,
+        help="Path to the private key file for certificate generation.")
+    add("-B", "--rsa-key-size", "--keysize", type=int, default=2048,
+        metavar="N", help="RSA key shall be sized N bits.")
+
+    add("-k", "--revoke", action="store_true", help="Revoke a certificate.")
+    add("-b", "--rollback", type=int, default=0, metavar="N",
+        help="Revert configuration N number of checkpoints.")
+    add("-v", "--view-config-changes", action="store_true",
+        help="View checkpoints and associated configuration changes.")
+    add("-r", "--redirect", action="store_true",
+        help="Automatically redirect all HTTP traffic to HTTPS for the newly "
+             "authenticated vhost.")
+
+    add("-e", "--agree-tos", dest="eula", action="store_true",
+        help="Skip the end user license agreement screen.")
+    add("-t", "--text", dest="use_curses", action="store_false",
+        help="Use the text output instead of the curses UI.")
+    add("--test", action="store_true", help="Run in test mode.")
+
+    # TODO: trailing slashes might be important! check and remove
+    add("--config-dir", default="/etc/letsencrypt/",
+        help="Configuration directory.")
+    add("--work-dir", default="/var/lib/letsencrypt/",
+        help="Working directory.")
+    add("--backup-dir", default="/var/lib/letsencrypt/backups/",
+        help="Configuration backups directory.")
+    add("--temp-checkpoint-dir",
+        default="/var/lib/letsencrypt/temp_checkpoint/",
+        help="Temporary checkpoint directory.")
+    add("--in-progress-dir",
```

Comment:
Okay, I'll move out `rev_tokens_dir`, `cert_key_backup`, `-in-rogress-dir` and `temp_checkpoint_dir` somewhere else (any ideas where?)... and compute them dynamically based on `IConfig.work_dir`.

I totally did not get your point about certificates, though. Do you mean `--cert-path` and `--cert-dir`? Do you want to specify filename or full path?